test/e2e/node/stress_probe.go

```diff
@@ -0,0 +1,196 @@
+
+const (
+	probeTestInitialDelaySeconds = 15
+	numTestPods                  = 1
+	numContainers                = 100
+	containerPort                = 2000
+)
+
+var _ = SIGDescribe("[Serial] [Slow] [Disruptive] Stress test probes", func() {
+	f := framework.NewDefaultFramework("stress-test-probe")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+
+	/*
+		Release: v1.28
+		Testname: Pod liveness probe stress test, using http endpoint, no restart
+		Description: A Pod is created with MANY containers with liveness probe on http endpoint /. Liveness probe MUST not fail to check health and the restart count should remain 0 for all containers.
+	*/
+	ginkgo.It("should *not be* restarted with http liveness probe stress test with MANY containers", func(ctx context.Context) {
+		livenessProbe := v1.Probe{
+			ProbeHandler:        httpGetHandler("/", containerPort),
+			InitialDelaySeconds: 15,
+			FailureThreshold:    1,
+		}
+		ports := []v1.ContainerPort{{ContainerPort: int32(containerPort)}}
+		probeArgs := []string{"serve-hostname", "--http", "--port", ""}
+		pod := livenessPodSpec("http", f.Namespace.Name, livenessProbe, ports, probeArgs...)
+		runLivenessTest(ctx, f, pod)
+	})
+
+	/*
+		Release: v1.28
+		Testname: Pod liveness probe stress test, using tcp socket, no restart
+		Description: A Pod is created with MANY containers with liveness probe on tcp socket 8080. Liveness probe MUST not fail to check health and the restart count should remain 0 for all containers.
+	*/
+	ginkgo.It("should *not* be restarted with a tcp:8080 liveness probe stress test with MANY containers", func(ctx context.Context) {
+		livenessProbe := v1.Probe{
+			ProbeHandler:        tcpSocketHandler(containerPort),
+			InitialDelaySeconds: 15,
+			FailureThreshold:    1,
+		}
+		ports := []v1.ContainerPort{{ContainerPort: int32(containerPort)}}
+		probeArgs := []string{"serve-hostname", "--tcp", "--port", "", "--http=false"}
+		pod := livenessPodSpec("tcp", f.Namespace.Name, livenessProbe, ports, probeArgs...)
+		runLivenessTest(ctx, f, pod)
+	})
+
+	/*
+		Release: v1.28
+		Testname: Pod liveness probe stress test, using grpc call, no restart
+		Description: A Pod is created with MANY containers with liveness probe on grpc service. Liveness probe MUST not fail to check health and the restart count should remain 0 for all containers.
+	*/
+	ginkgo.It("should *not* be restarted with a GRPC liveness probe stress test with MANY containers", func(ctx context.Context) {
+		livenessProbe := v1.Probe{
+			ProbeHandler: v1.ProbeHandler{
+				GRPC: &v1.GRPCAction{
+					Port:    containerPort,
+					Service: nil,
+				},
+			},
+			InitialDelaySeconds: probeTestInitialDelaySeconds,
+			TimeoutSeconds:      5, // default 1s can be pretty aggressive in CI environments with low resources
+			FailureThreshold:    1,
+		}
+		ports := []v1.ContainerPort{{ContainerPort: int32(containerPort)}}
+		probeArgs := []string{"grpc-health-checking", "--port", "", "--http-port", ""}
```

Comment:
I think that you can just add it entirely in `newContainer` based on the type, can't you?

This logic is very brittle since it depends on the positional arguments and it goes through two other functions later that fill in the gaps, there has to be a more readable way ,  can't we just add the entire command/args inside `newContainer`? is there any reason why we have to declare them here?

Reply:
yeah, I will try to change logic here.